server/controllers/prisonerCellHistoryController.ts

```diff
@@ -0,0 +1,168 @@
+      const queryParams: PagedListQueryParams = {}
+
+      const { bookingId, firstName, middleNames, lastName } = prisonerData
+      const name = formatName(firstName, middleNames, lastName, { style: NameFormatStyle.firstLast })
+
+      this.prisonApiClient = this.prisonApiClientBuilder(clientToken)
+
+      const page = 0
+      const cells = await this.prisonApiClient.getOffenderCellHistory(bookingId, { page, size: 10000 })
+      const agencyData = await this.prisonApiClient.getAgencyDetails((await cells).content[page].agencyId)
+      const staff = await Promise.all(
+        (await cells).content.map(cell => this.prisonApiClient.getStaffDetails(cell.movementMadeBy)),
+      )
+
+      const cellData = cells.content.map(cell => {
+        const staffDetails = staff.find(user => cell.movementMadeBy === user.username)
+        const agencyName = cell.agencyId
+        const agency = agencyData
+        const agencyDescription = agency ? agency.description : null
+
+        return {
+          establishment: agencyDescription,
+          location: extractLocation(cell.description, agencyName),
+          isTemporaryLocation: isTemporaryLocation(cell.description),
+          movedIn: cell.assignmentDateTime && formatTimestampToDateTime(cell.assignmentDateTime),
+          movedOut: cell.assignmentEndDateTime && formatTimestampToDateTime(cell.assignmentEndDateTime),
+          assignmentDateTime: moment(cell.assignmentDateTime).format('YYYY-MM-DDTHH:mm:ss'),
+          assignmentEndDateTime: cell.assignmentEndDateTime
+            ? moment(cell.assignmentEndDateTime).format('YYYY-MM-DDTHH:mm:ss')
+            : undefined,
+          livingUnitId: cell.livingUnitId,
+          agencyId: agencyName,
+          movedInBy: formatName(staffDetails.firstName, '', staffDetails.lastName),
+        }
+      })
+
+      const cellDataLatestFirst = cellData.sort((left, right) =>
+        sortByDateTime(right.assignmentDateTime, left.assignmentDateTime),
+      )
+      const currentLocation = cellDataLatestFirst.slice(0, 1)[0]
+      const occupants =
+        (currentLocation && (await this.prisonApiClient.getInmatesAtLocation(currentLocation.livingUnitId, {}))) || []
+
+      const previousLocations = cellDataLatestFirst.slice(1)
+
+      const prisonerProfileUrl = `/prisoner/${offenderNo}`
+
+      if (req.query.page) queryParams.page = +req.query.page
+      if (req.query.sort) queryParams.sort = req.query.sort as string
+      if (req.query.alertType) queryParams.alertType = req.query.alertType as string[]
+      if (req.query.from) queryParams.from = req.query.from as string
+      if (req.query.to) queryParams.to = req.query.to as string
+      if (req.query.showAll) queryParams.showAll = Boolean(req.query.showAll)
+
+      // Render page
+      return res.render('pages/prisonerCellHistoryPage', {
+        pageTitle: 'Location details',
+        ...mapHeaderData(prisonerData, inmateDetail, res.locals.user, 'location-details'),
```

Comment:
This is mostly used for the banner on the overview. There's a new method `mapHeaderNoBannerData` for pages where there is no banner, which is more efficient and doesn't require `inmateDetail`, so you might find you don't need that object at all in this controller.